### PR TITLE
Change default captcha difficulty to easy

### PR DESCRIPTION
**Pull request by weex**
Fixes visual part of #1640. Changes in docker/federation maybe not necessary but included for completeness.

### CODE REVIEW

File: crates/api/src/local_user.rs [hunk 1]
```diff
@@ -125,7 +125,7 @@ impl Perform for GetCaptcha {
       "easy" => gen(Difficulty::Easy),
       "medium" => gen(Difficulty::Medium),
       "hard" => gen(Difficulty::Hard),
-      _ => gen(Difficulty::Medium),
+      _ => gen(Difficulty::Easy),
```
Comment (Nutomic) [inline crates/api/src/local_user.rs:128]: You could change this to Difficulty::default(), then theres one less place to worry about.

File: docker/federation/lemmy_alpha.hjson [hunk 2]
```diff
@@ -22,7 +22,7 @@
   }
   captcha: {
     enabled: false
-    difficulty: medium
+    difficulty: easy
```
Comment (Nutomic) [inline docker/federation/lemmy_alpha.hjson:25]: This doesnt matter at all, the captcha is already disabled above and anyway its only for testing.

**Comment by Nutomic** [comment]
As for the PR itself, I didnt notice much of a change in difficulty between easy or medium, both times I was able to solve captchas quite easily. But then I tried solving it in lowercase (as many people probably do), and suddenly it kept failing. So what we should do is either add a message in lemmy-ui that the captcha is case sensitive, or or make a change in the backend to make it ignore case. The first option seems better, because otherwise the captcha would likely get easier for bots. 

So would you like to make a PR for that instead?

File: crates/api/src/local_user.rs (hunk 1 above)
Comment (dessalines) [reply crates/api/src/local_user.rs:128]: That doesn't exist there, these enums are from the captcha crate (Only easy, medium, and hard). So `Difficulty::Easy` is correct here.

**Comment by dessalines** [comment]
The easiest way would be to send both to lowercase in the comparison, that'd be here:

https://github.com/lemmynet/lemmy/blob/clippy_fix_1/crates/websocket/src/handlers.rs#L252

**Comment by weex** [comment]
Thanks for the feedback! I prefer the to_lowercase() option so will add that and remove the unnecessary test param change.

**Comment by Nutomic** [comment]
I suggest that you simply make a new PR for that (and close this one).

**Comment by weex** [comment]
Closing in lieu of https://github.com/LemmyNet/lemmy/pull/1649

**Pull request closed without merging**